httpbp/middlewares.go

```diff
@@ -323,9 +326,13 @@ func (r *statusCodeRecorder) WriteHeader(code int) {
 }
 
 func (r *statusCodeRecorder) getCode(err error) int {
-	if r.code != 0 {
+	return processCodeErr(err, r.code)
+}
+
+func processCodeErr(err error, code int) int {
```

Comment:
This name has nagged at me but I haven't really come up with why.  Maybe an API like `errorCodeForMetrics(explicitCode int, requestError err) int` would read more clearly?